hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -383,28 +314,18 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 			return ctrl.Result{}, fmt.Errorf("failed to get ignition CA secret: %w", err)
 		}
 	}
-	tokenSecret := ignitionserver.IgnitionTokenSecret(controlPlaneNamespace)
-	if err := r.Get(ctx, ctrlclient.ObjectKeyFromObject(tokenSecret), tokenSecret); err != nil {
-		if apierrors.IsNotFound(err) {
-			meta.SetStatusCondition(&nodePool.Status.Conditions, metav1.Condition{
-				Type:               string(hyperv1.IgnitionEndpointAvailable),
-				Status:             metav1.ConditionFalse,
-				Reason:             hyperv1.IgnitionTokenMissingReason,
-				ObservedGeneration: nodePool.Generation,
-			})
-			r.Log.Info("still waiting for ignition token secret to exist")
-			return ctrl.Result{}, nil
-		} else {
-			return ctrl.Result{}, fmt.Errorf("failed to get token secret: %w", err)
-		}
-	}
 
 	if result, err := controllerutil.CreateOrUpdate(ctx, r.Client, userDataSecret, func() error {
+		if userDataSecret.Annotations == nil {
+			userDataSecret.Annotations = make(map[string]string)
+		}
+		userDataSecret.Annotations[nodePoolAnnotation] = ctrlclient.ObjectKeyFromObject(nodePool).String()
```

Comment:
Also set an owner ref?

Reply:
who would be the owner since the NodePool live in a different ns?